programmers-coding-test/sangjun/greedy/1이될때까지.js

```diff
@@ -0,0 +1,18 @@
+function untilOne(first, second) {
+    let result = 0;
+    
+    while (first > 1) {
+        if (first % second === 0) {
+            first = first / second;
+            result += 1;
+        } else {
+            first -= 1;
+            result += 1;
+        }
+    }
```

Comment:
result -> count가 적합하다.
삼항 연산으로 식을 간추릴 수 있으며, count는 어떤 조건이든 증감시켜주는 것이므로 중복된다. 조건 밖으로 빼주는 것이 맞다.
```suggestion
function untilOne(n, k) {
    let count = 0;
    
    while (n > 1) {
        if (n % k=== 0) n /= k; // 같은 항과 연산할 경우 “/=“ 연산자를 쓰도록 하자.
        else first -= 1;
        count++;
    }
  
  
/*
 * 삼항 연산으로 바꾸고, count는 조건 밖으로 뺐다.
 */

let cnt = 0;
while (N !== 1) {
  N = N % K ? N - 1 : N / K;  // N이 K로 나눠지지 않으면 N에서 1을 빼고, 나눠지면 N / K를 수행하여 N에 대입한다.
  cnt++;
}

```